.github/workflows/dp-workflow.yml

```diff
@@ -6,7 +6,7 @@ on:  # yamllint disable-line rule:truthy
     branches:
       - master
       - 'v1.*'
-  pull_request_target:
+  pull_request:
```

Comment:
this reverts the change suggested in https://github.com/magma/magma/issues/12855#issuecomment-1149883837. Can you elaborate why this was done? I think @Neudrino had some reason for his suggestion

Reply:
The suggestion was to use the `pull_request_target` trigger action to make secrets available. We took the advice without actually verifying the validity of the suggestion.
`pull_request_target` is documented as follows:
```
This event runs in the context of the base of the pull request, rather than in the context of the merge commit, as the pull_request event does. This prevents execution of unsafe code from the head of the pull request that could alter your repository or steal any secrets you use in your workflow. This event allows your workflow to do things like label or comment on pull requests from forks. Avoid using this event if you need to build or run code from the pull request.
```
We are building code from the pull request, as that is the whole idea of running integration tests.

Secondly, the only secret we use is the `GITHUB_TOKEN` which is an exception to availability of secrets in PRs from forks:
```
With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.
```

Now, since we have introduced the `pull_request_target`, the CI was run from the base of the PRs, rather from the head of the PR branch, and that hid the mistake in the `Makefile`.